minifiers/minifiers.go

```diff
@@ -69,47 +83,73 @@ func New(mediaTypes media.Types, outputFormats output.Formats, cfg config.Provid
 	}
 
 	// We use the Type definition of the media types defined in the site if found.
-	if !conf.DisableCSS {
-		addMinifier(m, mediaTypes, "css", &conf.Tdewolff.CSS)
-	}
-	if !conf.DisableJS {
-		addMinifier(m, mediaTypes, "js", &conf.Tdewolff.JS)
-		m.AddRegexp(regexp.MustCompile("^(application|text)/(x-)?(java|ecma)script$"), &conf.Tdewolff.JS)
-	}
-	if !conf.DisableJSON {
-		addMinifier(m, mediaTypes, "json", &conf.Tdewolff.JSON)
-		m.AddRegexp(regexp.MustCompile(`^(application|text)/(x-|(ld|manifest)\+)?json$`), &conf.Tdewolff.JSON)
-	}
-	if !conf.DisableSVG {
-		addMinifier(m, mediaTypes, "svg", &conf.Tdewolff.SVG)
-	}
-	if !conf.DisableXML {
-		addMinifier(m, mediaTypes, "xml", &conf.Tdewolff.XML)
-	}
+	addMinifier(m, mediaTypes, "css", getMinifier(conf, "css"))
+
+	addMinifier(m, mediaTypes, "js", getMinifier(conf, "js"))
+	m.AddRegexp(regexp.MustCompile("^(application|text)/(x-)?(java|ecma)script$"), getMinifier(conf, "js"))
+
+	addMinifier(m, mediaTypes, "json", getMinifier(conf, "json"))
+	m.AddRegexp(regexp.MustCompile(`^(application|text)/(x-|(ld|manifest)\+)?json$`), getMinifier(conf, "json"))
+
+	addMinifier(m, mediaTypes, "svg", getMinifier(conf, "svg"))
+
+	addMinifier(m, mediaTypes, "xml", getMinifier(conf, "xml"))
 
 	// HTML
-	if !conf.DisableHTML {
-		addMinifier(m, mediaTypes, "html", &conf.Tdewolff.HTML)
-		for _, of := range outputFormats {
-			if of.IsHTML {
-				m.Add(of.MediaType.Type(), &conf.Tdewolff.HTML)
-			}
+	addMinifier(m, mediaTypes, "html", getMinifier(conf, "html"))
+	for _, of := range outputFormats {
+		if of.IsHTML {
+			m.Add(of.MediaType.Type(), getMinifier(conf, "html"))
 		}
 	}
 
 	return Client{m: m, MinifyOutput: conf.MinifyOutput}, nil
 }
 
-func addMinifier(m *minify.M, mt media.Types, suffix string, min minify.Minifier) {
-	types := mt.BySuffix(suffix)
-	for _, t := range types {
-		m.Add(t.Type(), min)
+// getMinifier returns the appropriate minify.MinifierFunc for the MIME
+// type suffix s, given the config c.
+func getMinifier(c minifyConfig, s string) minify.Minifier {
+	minifiersForSuffixSettings := map[string]struct {
```

Comment:
I know this isn't a hot path, but my head twisted a little when I see a map like this with values that just get thrown away after selecting one. 

I would prefer a simpler switch statement e.g.

```go
case "css" && !c.DisableCSS:
   return &c.Tdewolff.CSS
...
default
   return noop
```